src/clients/inst_kickoff.rb

```diff
@@ -46,15 +46,18 @@ def main
         )
       end
 
-      # copy the credential files (libzypp loads them from target)
-      # and the repository cache to the target system
-      Pkg.SourceCacheCopyTo(Installation.destdir)
-
-      # symlink the cache from inst-sys to save same space (RAM!)
-      cache_path = Pkg.ZConfig()["repo_cache_path"] || "/var/cache/zypp"
-      log.info("Zypp cache size: #{`du -h -s #{cache_path.shellescape}`}")
-      ::FileUtils.rm_rf(cache_path)
-      File.symlink(File.join(Installation.destdir, cache_path), cache_path)
+      # copy the zypp cache to the target system (only when running in inst-sys!)
+      if Stage.initial
+        # copy the credential files (libzypp loads them from target)
+        # and the repository cache to the target system
+        Pkg.SourceCacheCopyTo(Installation.destdir)
+
+        # symlink the cache from inst-sys to save same space (RAM!)
+        cache_path = Pkg.ZConfig()["repo_cache_path"] || "/var/cache/zypp"
+        log.info("Zypp cache size: #{`du -h -s #{cache_path.shellescape}`}")
+        ::FileUtils.rm_rf(cache_path)
+        File.symlink(File.join(Installation.destdir, cache_path), cache_path)
+      end
```

Comment:
NP: I'd move this kind of blocks to their own method, using a guard clause at the very top to avoid its execution if not running in the expected stage. You know, something like 

```ruby
# Copy the zypp cache to the target system (only when running in inst-sys!)
# 
# More documentation, if needed
# ...
#
def copy_zypp_cache
  return unless Stage.initial

  ...
end
```

For me is more readable and helps to document and understand the code :)